src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -281,6 +285,11 @@ module private Token =
           version = streamVersion + 1L
           streamBytes = -1 }
     let inline streamVersion (token : StreamToken) = unbox<int64> token.value
+    let shouldSnapshot batchSize prev next =
+        let previousVersion = prev.version
+        let nextVersion = next.version
+        let estimatedSnapshotPos = previousVersion - (previousVersion % batchSize)
+        nextVersion - estimatedSnapshotPos >= batchSize
```

Comment:
naive implementation, but should suffice fine here

Reply:
A bit of Maths™️  😅

We want to snapshot every `B`atchsize events, and version is function of the number of events in the stream

Given a version I can therefore know when the last snapshot occurred with a bit of modulo magic

`version - (version % batchsSize)`

e.g. if the batchsize is 10, and the version is 15 this yield `15 - (15 % 10) = 15 - 5 = 10`

 Now that we have the previous snapshot placement we know if we have to snapshot based on the difference between the snapshot version and the current version.

Reply:
>A bit of Maths™️ 😅

and there's absolutely definitely no way you could be off by one, ever?
and the previous version is definitely from the snapshot, and not from the last cached append (it scanned fine, but I'd be pressing * and # for a long time making sure!)

Reply:
> and there's absolutely definitely no way you could be off by one, ever?

the formula is sound. But I cannot guarantee absence of off-by-ones. I could be off by one with a `>` vs `>=`, but I think that would've been caught in the integration test (with an additional slice read being necessary)

Reply:
Yes, not doubting you or the formula, which seems correct and is obviously not the most mind bending math EVAR. Just asking the rhetorical question about whether such edge cases are covered. In general the cases you took from try to exercise snapshots being 'just' due or 'just not yet', but e.g. for ESDB, the token tests cover a lot of these cases.